packages/bot-web-ui/src/components/chart/chart.jsx

```diff
@@ -13,77 +14,96 @@ import '../../assets/sass/chart.scss';
 // import { symbolChange }      from '../../SmartChart/Helpers/symbol';
 // import AllMarkers            from '../../SmartChart/Components/all-markers.jsx';
 
-class Chart extends React.Component {
-    chartTopWidgets = () => <ChartTitle onChange={this.props.onSymbolChange} />;
+const ChartTopWidgets = ({ onChange }) => <ChartTitle onChange={onChange} />;
 
-    chartToolbarWidget = () => (
-        <ToolbarWidgets updateChartType={this.props.updateChartType} updateGranularity={this.props.updateGranularity} />
-    );
+const ChartToolbarWidget = ({ updateChartType, updateGranularity }) => (
+    <ToolbarWidgets updateChartType={updateChartType} updateGranularity={updateGranularity} />
```

Comment:
`ToolbarWidgets` could be passed directly to `SmartChart`, because both `ChartToolbarWidget` and `ToolbarWidgets` have the same signature.